src/Featurizers/ForecastingPivotFeaturizer.h

```diff
@@ -0,0 +1,129 @@
+            typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+            std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    ForecastingPivotTransformer(void) = default;
+    ForecastingPivotTransformer(Archive &ar);
+
+    ~ForecastingPivotTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ForecastingPivotTransformer);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using NullableType                      = typename Microsoft::Featurizer::Traits<InputT>::nullable_type;
+    using MatrixType                        = Microsoft::Featurizer::RowMajMatrix<NullableType>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        if (input.empty()) {
+            return;
+        }
+        
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type> ret;
+        Eigen::Index const col_size(input[0].cols());
+
+        for (Eigen::Index col = 0; col < col_size; ++col) {
+            bool has_nan(false);
+
+            // if any nans is found in the current column across all matrixes, this column is omitted
+            for (MatrixType const & matrix : input) {
+                // all matrixes must have the same column number
+                assert(matrix.cols() == col_size);
+
+                for (Eigen::Index row = 0; row < matrix.rows(); ++row) {
+                    if (Traits<NullableType>::IsNull(matrix(row, col))) {
+                        has_nan = true;
+                        break;
+                    }
+
+                    ret.emplace_back(matrix(row, col));
+                }
+
+                if (has_nan) {
+                    break;
+                }
+            }
+
+            if (!has_nan) {
+                callback(ret);
+            }
+
+            ret.clear();
+        }
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \typedef       ForecastingPivotEstimator
+///  \brief         Estimator that creates `ForecastingPivotTransformer`.
+///
+template <typename T>
+using ForecastingPivotEstimator           = Components::InferenceOnlyEstimatorImpl<ForecastingPivotTransformer<T>>;
```

Comment:
Unfortunately, we have to give a static name to the Estimator; this means that we can't directly use the `InferenceOnlyEstimatorImpl` object. This is annoying and unintuitive, so I'd love to find a better way to pass a string in this scenario.

Reply:
Gotcha! Change the `using` clause to inherit from `InferenceOnlyEstimatorImpl`.